pkg/scheduler/framework/plugins/defaultpreemption/default_preemption.go

```diff
@@ -186,29 +193,28 @@ func (pl *DefaultPreemption) getOffsetAndNumCandidates(numNodes int32) (int32, i
 
 // FindCandidates calculates a slice of preemption candidates.
 // Each candidate is executable to make the given <pod> schedulable.
-func (pl *DefaultPreemption) FindCandidates(ctx context.Context, state *framework.CycleState, pod *v1.Pod, m framework.NodeToStatusMap) ([]Candidate, error) {
+func (pl *DefaultPreemption) FindCandidates(ctx context.Context, state *framework.CycleState, pod *v1.Pod, m framework.NodeToStatusMap) ([]Candidate, int, framework.NodeToStatusMap, error) {
 	allNodes, err := pl.fh.SnapshotSharedLister().NodeInfos().List()
 	if err != nil {
-		return nil, err
+		return nil, 0, nil, err
 	}
 	if len(allNodes) == 0 {
-		return nil, fmt.Errorf("no nodes available")
+		return nil, 0, nil, fmt.Errorf("no nodes available")
 	}
-
-	potentialNodes := nodesWherePreemptionMightHelp(allNodes, m)
+	potentialNodes, unschedulableNodeStatus := nodesWherePreemptionMightHelp(allNodes, m)
 	if len(potentialNodes) == 0 {
 		klog.V(3).Infof("Preemption will not help schedule pod %v/%v on any node.", pod.Namespace, pod.Name)
 		// In this case, we should clean-up any existing nominated node name of the pod.
 		if err := util.ClearNominatedNodeName(pl.fh.ClientSet(), pod); err != nil {
 			klog.Errorf("Cannot clear 'NominatedNodeName' field of pod %v/%v: %v", pod.Namespace, pod.Name, err)
 			// We do not return as this error is not critical.
 		}
-		return nil, nil
+		return nil, len(allNodes), unschedulableNodeStatus, nil
```

Comment:
Do we need the length of all nodes? as all other PostFilter plugins also know that.

Reply:
yes, I need to build a obj of `FitError` where how many node are evaluated are needed there.